docs/architecture-decisions/0013-use-sqs-and-airflow-for-task-execution.md

```diff
@@ -0,0 +1,25 @@
+# 13. Use SQS and Airflow for Task Execution
+
+Date: 2020-04-03
+
+## Status
+
+Accepted
+
+Supercedes [12. Use Lambda and Fargate for Task Execution](0012-use-lambda-and-fargate-for-task-execution.md)
+
+## Context
+
+The execution model described by ADR 12 was designed before we had Airflow. It works, but we'd like to simplify things by moving it to Airflow to avoid having similar processes handled in different ways.
+
+## Decision
+
+We will change the S3 notification from triggering a Lambda to sending a message to an SQS queue. We will configure a single workflow in Airflow that begins with an SQS sensor.
+
+## Consequences
+
+1. Instead of having a different indexing process for each source, there will be a single indexing workflow in Airflow that gets run on every file uploaded to the S3 bucket. Mario will need logic added to it to handle correctly indexing based only on the name of the bucket and key.
```

Comment:
Just clarifying here to make sure I understand. Both of our workflows will need to adjust. Aleph gets rewritten to move from the lambda to this new process described here.

Aspace workflow will end when it writes to the bucket (which will be configured to update SQS) and a separate job that monitors that SQS will pick it up from there and complete it.

Is that correct?

Reply:
Yeah, that's correct.